### PR TITLE
Switch to version tags with a leading 'v'

### PR DESCRIPTION
**Pull request by kflynn**
This way, Go code using the protobuf code we generate can reference `v0.77.0` or the like, instead of being forced to use a commit hash.

(Yes, Go is silly.)

Note that the version numbers in the diag UI will stay the same (e.g. `0.77.0` instead of `v0.77.0`).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,9 +69,9 @@ IS_PRIVATE ?= $(findstring private,$(_git_remote_urls))
 # Note that for everything except RC builds, VERSION will be set to the version
 # we'd use for a GA build. This is by design.
 ifneq ($(GIT_TAG_SANITIZED),)
-VERSION = $(shell printf "$(GIT_TAG_SANITIZED)" | sed -e 's/-.*//g')
+VERSION = $(shell printf "$(GIT_TAG_SANITIZED)" | sed -e 's/-.*//g' | sed -e 's/^[vV]//')
 else
```
Comment (LukeShu) [inline Makefile]: I'd have `$(patsubst v%,%, …)`.

Comment (LukeShu) [reply Makefile]: Actually, `VERSION = $(patsubst v%,%,$(firstword $(subst -, ,$(GIT_TAG_SANITIZED)))`

File: Makefile [hunk 2]
```diff
@@ -69,9 +69,9 @@ IS_PRIVATE ?= $(findstring private,$(_git_remote_urls))
...
-VERSION = $(GIT_VERSION)
+VERSION = $(shell printf "$(GIT_VERSION)" | sed -e 's/^[vV]//')
```
Comment (LukeShu) [inline Makefile]: Likewise.

File: Makefile (hunk 1 above)
Comment (kflynn) [reply Makefile]: I'm going to land this as-is and then make the change after 0.77.0.

**Pull request merged**